bitnami/etcd/templates/statefulset.yaml

```
@@ -25,9 +25,11 @@ spec:
         {{- if .Values.podAnnotations }}
         {{- include "etcd.tplValue" ( dict "value" .Values.podAnnotations "context" $) | nindent 8 }}
         {{- end }}
-        {{- if .Values.metrics.podAnnotations }}
+        {{- if and .Values.metrics.enabled .Values.metrics.podAnnotations }}
+        {{- if not (and .Values.metrics.serviceMonitor .Values.metrics.serviceMonitor.enabled) }}
```

Wouldn't it be the same than the following?
```suggestion
        {{- if and .Values.metrics.enabled .Values.metrics.podAnnotations ( not .Values.metrics.serviceMonitor.enabled ) }}
```

If you change this, you should remove the `{{ - end }}` below.

---

Line 23 should be updated as well, right? There could be situations in which the annotations would be empty, i.e.:

- `metrics.enabled=true`
- `metrics.serviceMonitor.enabled=true`

---

Yes indeed if miss that the `metrics.serviceMonitor` is present in the default values.
And yes line 23 should be updated